core/src/main/java/org/hibernate/search/jsr352/internal/steps/lucene/EntityReader.java

```diff
@@ -242,7 +242,8 @@ private ScrollableResults buildScrollUsingHQL(StatelessSession ss, String HQL) {
 	private ScrollableResults buildScrollUsingCriteria(StatelessSession ss,
 			PartitionBound unit, Object checkpointId, JobContextData jobData) {
 
-		String idName = MassIndexerUtil.getIdName( entityType, session );
+		String idName = Projections.id().toString();
```

Comment:
> Using toString() for anything else than debugging seems like a bad idea to me in general ...

I don't have other choice than using the `toString()` method. You suggested me to use `Projections#id()`, but it returns an `IdentifierProjection`, which is not usable for creating an object `org.hibernate.criterion.Order`. All constructors of `Order` requires a String as argument, `IdentifierProjection` is not a String so it doesn't fit the constructors.

> ... Maybe you could test that?

I don't understand the second paragraph. Do you want me to create an entity type `MyEntity.class` which has a customized name of the identifier and test it through the batch indexing job?

```java
@Entity
@Indexed
public class MyEntity implements Serializable {

    @Id
    @DocumentId
    private int customId;  // sth like this: 
                           // id column is named sth other than id?

    ...
}
```

As for the 3re paragraph,

- If the Alias were created using method `Projections#alias(Project, String)` as you suggested, the `toString()` method can be avoid. But how to use this alias? Order by alias does not exists. No matter we're constructing an ascending order `org.hibernate.criterion.Order#asc(String)` or a descending order `org.hibernate.criterion.Order#desc(String)`, both of them use String as an argument and not a projection. `Order.property( "aliased_id" )` doesn't exist. (I'm using Hibernate ORM _5.1.0.Final_).
- If the Alias were created using method `org.hibernate.Criteria#createAlias(String, String)`, the `toString()` method is needed, because it requires a String argument.